src/utils/getUrlByRequestOptions.test.ts

```diff
@@ -117,19 +122,45 @@ it('supports "hostname" instead of "host" and "port"', () => {
   )
 })
 
-it('handles IPv6 hostnames', () => {
-  expect(
-    getUrlByRequestOptions({
-      host: '::1',
-      path: '/resource',
-    }).href
-  ).toBe('http://[::1]/resource')
+it('use "hostname" if both "hostname" and "host" are specified', () => {
+  const options: RequestOptions = {
+    protocol: 'https:',
+    host: 'host',
+    hostname: 'hostname',
+    path: '/resource',
+  }
 
-  expect(
-    getUrlByRequestOptions({
-      host: '::1',
-      port: 3001,
-      path: '/resource',
-    }).href
-  ).toBe('http://[::1]:3001/resource')
+  // TODO: we should ignore the port
+  expect(getUrlByRequestOptions(options).href).toBe(
+    'https://hostname/resource'
+  )
+})
+
+describe('IP v6', () => {
```

Comment:
I recommend we add a separate test file for IPv6 instead of using `describe()`.

Reply:
Sure!
in `utils`? `ip6.test.ts`?

Reply:
We only have a couple of tests for IPv6, so how about this?

```js
it('parses "host" in IPv6')

it('parses "host" and "port" in IPv6')
```